lib/Fmt_odoc.ml

```diff
@@ -198,19 +228,27 @@ let rec fmt_inline_elements elements =
         in
         hovbox
           (1 + String.length s + 1)
-          (wrap_elements "{" "}" ~always_wrap:true (str_normalized s) elems)
+          (wrap_elements "{" "}" ~always_wrap:true (str_normalized c s) elems)
         $ aux t
     | `Reference (_kind, rf, txt) :: t ->
         let rf = wrap "{!" "}" (fmt_reference rf) in
         wrap_elements "{" "}" ~always_wrap:false rf txt $ aux t
     | `Link (url, txt) :: t ->
-        let url = wrap "{:" "}" (str_normalized url) in
+        let url = wrap "{:" "}" (str_normalized c url) in
         hovbox 2 @@ wrap_elements "{" "}" ~always_wrap:false url txt $ aux t
   in
   aux (List.map elements ~f:(ign_loc ~f:Fn.id))
 
-and fmt_nestable_block_element c = function
-  | `Paragraph elems -> fmt_inline_elements elems
+and fmt_nestable_block_element c {Loc.location= l; value} =
+  match value with
+  | `Paragraph elems ->
+      if c.ocp_indent_compat then
+        match c.input with
+        | Some input ->
+            let lines = lines_of_span l input in
+            vbox 0 (list lines "@," str)
```

Comment:
Why is the `input` needed ? Doesn't `` `Space `` nodes contain newlines ?

Reply:
I've cleaned up the code and now relying only on these spaces.